File: src/LeetCode/Algorithms/JavaScript/Two_Sum.js
/**
 * https://leetcode.com/problems/two-sum/
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
	const obj = {};
	
	for (let i = 0; i < nums.length; i++) {
		let key = target - nums[i];
		
		if (obj[key] !== undefined) {
			return [obj[key], i];
		}
		
		obj[nums[i]] = i;
	}
	
	/*const obj = new Map();
	
	for (let i = 0; i < nums.length; i++) {
		let key = target - nums[i];
		
		if (obj.has(key)) {
			return [obj.get(key), i];
		}
		
		obj.set(nums[i], i);
	}*/
};

let nums = [2, 7, 11, 15];
let target = 9;

console.log(twoSum(nums, target));

File: src/LeetCode/Algorithms/Java/Two_Sum.java
package LeetCode.Algorithms.Java;

import java.util.HashMap;

// https://leetcode.com/problems/two-sum/
public class Two_Sum {
	public static void main(String[] args) {
		int[] nums = {-1, -2, -3, -4, -5}; // {3, 2, 4} // {3, 3} // {0, 4, 3, 0} // {-3, 4, 3, 90}
		int target = -8;

		int[] result = twoSum(nums, target);

		for(int x : result) {
			System.out.print(x + " ");
		}
	}

	public static int[] twoSum(int[] nums, int target) {
		HashMap<Integer, Integer> hashMap = new HashMap<Integer, Integer>();

		for(int i = 0; i < nums.length; i++)
			hashMap.put(nums[i], i);

		for(int i = 0; i < nums.length; i++) {
			int diff = target - nums[i];
			Integer value = hashMap.get(diff);

			if(value != null && !value.equals(i))
				return new int[]{i, value};
		}

		return new int[]{};
	}
}